vertical-pod-autoscaler/pkg/recommender/util/histogram_options.go

```diff
@@ -109,25 +111,30 @@ func (o *exponentialHistogramOptions) NumBuckets() int {
 	return o.numBuckets
 }
 
+// Returns the index of the bucket for given value. This is the inverse function to
+// GetBucketStart(), which yields the following formula for the bucket index:
+//    bucket(value) = floor(log(value/firstBucketSize*(ratio-1)+1) / log(ratio))
 func (o *exponentialHistogramOptions) FindBucket(value float64) int {
 	if value < o.firstBucketSize {
 		return 0
 	}
-	bucket := int(math.Log(value/o.firstBucketSize)/math.Log(o.ratio)) + 1
+	bucket := int(math.Log(value/o.firstBucketSize*(o.ratio-1)+1) / math.Log(o.ratio))
```

Comment:
These formulas got pretty complex. I would recommend splitting to a few statements with named subparts or even providing a separate short function to calculate the bucket number.